forest.hs

```diff
@@ -1,40 +1,41 @@
-{-# LANGUAGE BangPatterns #-}
-import Data.List
+module Main where
+
+import qualified Data.Set as S
 import System.Environment
 
 data Forest = Forest
   { goats :: !Int
   , wolves :: !Int
   , lions :: !Int
```

Comment:
Ah, I see. I tried taking it out and performance dropped by a almost factor of two, so definitely worth keeping :)